crates/ruma-common/src/events/room/message.rs

```diff
@@ -467,15 +468,18 @@ impl From<RoomMessageEventContent> for MessageType {
 #[derive(Clone, Debug)]
 #[allow(clippy::manual_non_exhaustive)]
 #[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
-pub enum Relation {
+pub enum Relation<C>
+where
+    C: EventContentWithoutRelation,
+{
```

Comment:
Why is this generic over the "original" content type that can hold a relation, rather than the relation-less one (then we wouldn't need a `where` clause and `Replacement` would just contain `Replacement<C>`.